Kernel/Memory/Heap.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * @file Heap.cpp
+ * @author Keeton Feavel (keeton@xyr.is)
+ * @brief
+ * @version 0.1
+ * @date 2022-01-24
+ *
+ * @copyright Copyright the Xyris Contributors (c) 2022
+ *
+ */
+#include "Heap.hpp"
+#include "paging.hpp"
+#include <Arch/Memory.hpp>
+#include <Library/rand.hpp>
+#include <Library/LinkedList.hpp>
+#include <Locking/Mutex.hpp>
+#include <Panic.hpp>
+
+#define HEAP_MAGIC  0x0B1E55ED
+#define HEAP_DEATH  0xBADA110C
+
+#define ALIGNMENT   16 // Memory byte alignment
+#define ALIGN_TYPE  uint8_t
+#define ALIGN_INFO  sizeof(ALIGN_TYPE) * 16 // Size of alignment info
+
+class Minor;
+
+class Major : public LinkedList::Node {
+public:
+    void initialize(size_t pages)
```

Comment:
Why isn't this a constructor?

Reply:
Because we have to request pages manually (no new or malloc by this point) so we have to manually cast the memory returned into a class.